core/pathresolver.go

```diff
@@ -16,11 +18,18 @@ var ErrTooManyLinks = errors.New("exceeded maximum number of links in ipns entry
 // Resolves the given path by parsing out /ipns/ entries and then going
 // through the /ipfs/ entries and returning the final merkledage node.
 // Effectively enables /ipns/ in CLI commands.
-func Resolve(n *IpfsNode, p path.Path) (*merkledag.Node, error) {
-	return resolveRecurse(n, p, 0)
+func Resolve(ctx context.Context, n *IpfsNode, p path.Path) (*merkledag.Node, error) {
+	r := resolver{ctx, n, p}
```

Comment:
It is already wrapped in (at least on the path from the http gateway I am currently working on). I think, this should already be setup by the commands lib when preparing the request. Starting to care about it down the road in some cases seems unwieldy.

I'd like to talk about these semantics (context setup, api surface) on tomorrow on/after the sprint meeting. We need a unified approach for this to polish the api properly.